src/Filecoin/Address.h

```diff
@@ -10,49 +10,79 @@
 
 #include <array>
 #include <cstdint>
-#include <string>
+#include <optional>
 #include <vector>
+#include <string>
 
 namespace TW::Filecoin {
 
 class Address {
   public:
+    /// The actor ID of the Ethereum Address Manager singleton.
+    static constexpr uint64_t ETHEREUM_ADDRESS_MANAGER_ACTOR_ID = 10;
+
     enum class Type : uint8_t {
         ID = 0,
         SECP256K1 = 1,
         ACTOR = 2,
         BLS = 3,
+        DELEGATED = 4,
         Invalid,
     };
 
-    /// Address data with address type prefix.
-    Data bytes;
+    /// Type of the Address.
+    Type type = Type::Invalid;
 
-    /// Determines whether a collection of bytes makes a valid address.
-    static bool isValid(const Data& data);
+    /// Actor ID.
+    /// This is used if `type` is either `ID` or `DELEGATED`.
+    uint64_t actorID = 0;
+
+    /// Address data payload (without prefixes and checksum).
+    Data payload;
+
+    /// Decodes `encoded` as a Filecoin address.
+    /// Returns `std::nullopt` on fail.
+    static std::optional<Address> fromBytes(const Data& encoded);
+
+    /// Parses `string` as a Filecoin address and validates the checksum.
+    /// Returns `std::nullopt` if `string` is not a valid address.
+    static std::optional<Address> fromString(const std::string& string);
```

Comment:
I recommend to have a type shortcut here: `using MaybeAddress = std::optional<Address>`